hack/make/run

```diff
@@ -48,39 +53,35 @@ if [ -n "$DOCKER_ROOTLESS" ]; then
 fi
 
 args="--debug \
-	--host "tcp://0.0.0.0:${listen_port}" --host "unix://${socket}" \
+	--host "tcp://0.0.0.0:${docker_listen_port}" --host "unix://${socket}" \
 	--storage-driver "${DOCKER_GRAPHDRIVER}" \
 	--userland-proxy="${DOCKER_USERLANDPROXY}" \
 	$storage_params \
 	$extra_params"
 
-echo "${dockerd} ${args}"
-if [ -n "$DELVE_PORT" ] && [ -z "$DOCKER_ROOTLESS" ]; then
-	# Run Docker daemon with debugger
-	echo "Exposing Delve debugger on port $DELVE_PORT"
-
-	# shellcheck disable=SC2086
-	exec dlv --listen=0.0.0.0:$DELVE_PORT \
+if [ -n "$DELVE_PORT" ]; then
+	dockerd="dlv --listen=0.0.0.0:$delve_listen_port \
 		--headless=true \
 		--log \
 		--api-version=2 \
 		--only-same-user=false \
 		--check-go-version=false \
 		--accept-multiclient \
-		exec "${dockerd}" -- ${args}
-elif [ -n "$DOCKER_ROOTLESS" ]; then
-	# Run the dockerd-rootless.sh script
-	# It will run Delve debugger when necessary
+		exec ${dockerd} --"
```

Comment:
So as a whole, this will be executed as;

```bash
exec dlv [delve options] exec dockerd -- [dockerd options]
```

Trying to get my head around that part (bear with me, perhaps I'm completely wrong, just glancing over);

- is there an `exec` too many there, or is that correct?
- doesn't the `--` terminal processing flags for `dockerd` ?


Basically, I was also looking if we could make the delve command a separate variable, so that it could all be called as;

```bash
exec ${delve} "${dockerd}" ${args}
```

or

```bash
${delve} exec "${dockerd}" ${args}
```

Reply:
I haven't looked at the delve docs, but this is probably `exec` the shell command and `exec` the delve subcommand (so both makes sense).

Reply:
oooh!

Reply:
Just as @tianon said, the first `exec` is from bash, while the second one is Delve's command to execute the precompiled binary (as opposed to building it with `dlv debug`).